manifest/docker_schema2.go

```diff
@@ -207,7 +209,23 @@ func (m *Schema2) UpdateLayerInfos(layerInfos []types.BlobInfo) error {
 	original := m.LayersDescriptors
 	m.LayersDescriptors = make([]Schema2Descriptor, len(layerInfos))
 	for i, info := range layerInfos {
-		m.LayersDescriptors[i].MediaType = original[i].MediaType
+		switch info.Compression {
```

Comment:
(Non-blocking, probably only theoretical at this point:

Should this only override known `MediaType` values? I have no idea, maybe not; it’s at least not immediately obvious.

Suppose someone uses an `com.example.org/completely-standards-violating-x-this-is-a-pretty-icon-for-the-container` MIME type and lists it in the manifest’s layer list; is it OK to just overwrite it because we decided to compress the layer? Arguably we should not have compressed a layer with an unknown MIME type in the first place…)

Reply:
I'll add a `// TODO` in the code. It sounds like a topic for another PR - potentially bigger since it brings up the question of which mime types does format X support and how and where do we enforce it.